lib/variant-model.js

```diff
@@ -45,17 +47,39 @@ _.extend(Variant.prototype, {
     return this;
   },
 
-  respondWithFile: function(fileName) {
+  respondWithFile: function(options) {
+    var self = this;
+
     return this.respondWith(function(request, reply) {
-      var self = this;
-      fileName = fileName.replace(/\{([^\}]+)\}/g, function(match, token) {
-        var val = request.params[token];
-        if (!val) {
-          val = self.state(token);
+      // Intentionally requiring smocks to lazy load options
+      var smocks = require('./index');
+
+      options = _.extend({
+        fileName: smocks.initOptions.fileName || 'default.json',
+        mockDir: smocks.initOptions.mockDir || './mocked-data',
+        statusCode: smocks.initOptions.statusCode || 200
+      }, options);
+
+      var pattern = /{\w+}\/*/g;
+      var path = self._route.path().replace(pattern, '');
+      var dataVariantFilePath = Path.join(options.mockDir, path, self.id() + '.json');
+
+      // Search for variant file
+      Fs.readFile(dataVariantFilePath, function (err, variantData) {
+        if (err) {
+          // No variant file found, search for the default json file
+          var dataDefaultFilePath = Path.join(options.mockDir, path, options.fileName);
+          Fs.readFile(dataDefaultFilePath, function (err, defaultData) {
+            if (err) {
+              return reply().code(404);
+            } else {
+              reply(defaultData).type('application/json').code(options.statusCode);
+            }
+          });
+        } else {
+          reply(variantData).type('application/json').code(options.statusCode);
         }
```

Comment:
Your suggestion to write following:

fs.readFile(filePath, function (err, data) {
  if (err) {
    return reply().code(404);
  }
  reply(data).type('application/json').code(options.statusCode);
});

Would make the code cleaner but we still need the case where you don't have a specific file for the variant i.e. variant2.json and want to use your default.json for all the variants which doesn't have a matching json file.

For example, with a request of /customer/123/ and variant of foo
1) /customer123/foo.json

If no variant json file is present in the directory structure
2)/customer/123/default.json (or what the user set in the options.fileName)